packages/flutter/lib/src/cupertino/colors.dart

```diff
@@ -80,3 +93,672 @@ class CupertinoColors {
   /// This is SystemRed in the iOS palette.
   static const Color destructiveRed = Color(0xFFFF3B30);
 }
+
+/// A [Color] subclass that represents a family of colors, and the currect effective
+/// color in the color family.
+///
+/// When used as a regular color, `CupertinoDynamicColor` is equivalent to the
+/// effective color, which is determined by the [BuildContext] it is last resolved
```

Comment:
I think it would be worth pointing out that the "effective" color is this color's [Color.value].